nucleus/modelci/eval_functions/base_eval_function.py

```diff
@@ -0,0 +1,56 @@
+import abc
+
+from ..constants import ThresholdComparison
+from ..data_transfer_objects.eval_function import (
+    EvalFunctionEntry,
+    EvaluationCriteria,
+)
+
+
+class BaseEvalFunction(abc.ABC):
+    """Abstract base class for concrete implementations of EvalFunctions
+
+    Operating on this class with comparison operators produces an EvaluationCriteria
+    """
+
+    def __init__(self, eval_func_entry: EvalFunctionEntry):
+        self.eval_func_entry = eval_func_entry
+
+    @classmethod
+    @property
+    @abc.abstractmethod
+    def name(cls) -> str:
+        """Name to look for in the EvalFunctionDefinitions"""
```

Comment:
Would it also make sense to have annother abstract method which returns the eval function id `ef_...`? I think this would be a nicer interface for the user, rather than to got through the client and list them first, as we have knowledge about the specific one here (or rather in the child class)